core/include/detray/utils/tuple_helpers.hpp

```diff
@@ -45,9 +46,33 @@ DETRAY_HOST_DEVICE inline constexpr decltype(auto) get(
     thrust::tuple<value_types...>& tuple) noexcept {
     return thrust::get<get_type_pos_v<query_t, value_types...>>(tuple);
 }
+
+template <std::size_t I, typename... value_types>
+DETRAY_HOST_DEVICE inline constexpr decltype(auto) get(
+    const detray::tuple<value_types...>& tuple) noexcept {
+    return detray::get<I>(tuple);
+}
+
+template <std::size_t I, typename... value_types>
+DETRAY_HOST_DEVICE inline constexpr decltype(auto) get(
+    detray::tuple<value_types...>& tuple) noexcept {
+    return detray::get<I>(tuple);
+}
```

Comment:
Wait... does this actually work right now with no non-const version of the called function existing...? If everything was set up "just so", I thought this version of the function could be pulled into this namespace with

```c++
using detray::get;
```

, as is done for `std::get` and `thrust::get`.

Also, the naming is a bit backward. Having the function that's specific to `detray::tuple` be named as `detray::get`, while the overloaded functions sit in `detray::detail::get` is a bit strange. :confused:

Reply:
This specific `detray::detail::get` function is fine. What worries me is that since there is no non-const version of `detray::get` at the moment, both versions of `detray::detail::get` will return a constant reference to the tuple elements. Whether or not the tuple itself was const or non-const.

My guess is that there is just no (test) code at the moment that would try to modify an element of a non-const tuple using `detray::get` or `detray::detail::get`...